src/main/java/com/google/cloud/tools/io/FileUtil.java

```diff
@@ -76,4 +77,38 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs)
           }
         });
   }
+
+  /**
+   * Implementation of recursive directory delete.
+   * https://docs.oracle.com/javase/7/docs/api/java/nio/file/FileVisitor.html
+   *
+   * @param target directory to delete
+   * @throws IOException if target is not a directory or underlying file mechanisms fail
+   */
+  public static void deleteDirectory(final Path target) throws IOException {
+    if (!Files.isDirectory(target)) {
+      throw new NotDirectoryException(target.toString());
+    }
+    Files.walkFileTree(
+        target,
+        new SimpleFileVisitor<Path>() {
```

Comment:
separate outer class please.
also see #314

Reply:
#314 was related to putting eclipse functionality in appengine-plugins-core, and even though this is now used in plugins-core, perhaps the point is still valid and these should just be moved into the classes using them and FileUtil should be removed.